classes/Event/class.xoctEvent.php

```diff
@@ -631,11 +631,52 @@ public function getAnnotationLink() {
 			if (xoctConf::getConfig(xoctConf::F_SIGN_ANNOTATION_LINKS)) {
 				$this->annotation_url = xoctSecureLink::sign($url);
 			} else {
-
 				$this->annotation_url = $url;
 			}
 		}
 
+		// If Annnotation token security is enabled
+		if(xoctConf::getConfig(xoctConf::F_ANNOTATION_TOKEN_SEC))  {
+			// Get Media URL
+			$media_object= $this->getFirstPublicationMetadataForUsage(xoctPublicationUsage::find(xoctPublicationUsage::USAGE_ANNOTATE))->getMedia();	
+			$media_url = $media_object[0]->url;
+
+			if (xoctConf::getConfig(xoctConf::F_SIGN_PLAYER_LINKS)) {			
+				//get duration and calculate expire date
+				$duration = $media_object[0]->duration;
+				
+				$valid_until = null;
+				if (xoctConf::getConfig(xoctConf::F_SIGN_PLAYER_LINKS_OVERWRITE_DEFAULT)) {
+					$duration_in_seconds = $duration / 1000;
+					$additional_time_percent = xoctConf::getConfig(xoctConf::F_SIGN_PLAYER_LINKS_ADDITIONAL_TIME_PERCENT) / 100;
+					$valid_until = gmdate("Y-m-d\TH:i:s\Z", time() + $duration_in_seconds + $duration_in_seconds * $additional_time_percent);
+				}
+				// Sign the url and parse the variables.
+				$media_url_signed = xoctSecureLink::sign($media_url, $valid_until);
+				$media_url_query = parse_url($media_url_signed, PHP_URL_QUERY);
+				$media_url = $media_url . '&' . $media_url_query;
+			}
+
+			// Get user and course ref id
+			$ref_id = $_GET['ref_id'];
```

Comment:
This is not the course's ref id, it's the ref id of the Opencast object (which is inside a course). But I guess that's ok?

Reply:
Ohh really? I didn't know that is the ref id from the opencast object. Well, it's a surprise, but a welcome one. 

It's ok, even better, because even if the user have access to the course, can't access the annotation tool if the teacher has not given access to the opencast object itself.